src/stdatamodels/jwst/datamodels/wcs_ref_models.py

```diff
@@ -398,6 +399,107 @@ def to_fits(self):
+    The model is for the specwcs for LRS Fixed Slit and LRSSlitless
+    Parameters
+    ----------
+    x_ref : float
+         x coordinate of reference position of fixed slit aperture
+    y_ref : float
+        y coordinate of reference position of fixed slit aperture
+    x_ref_slitless : float
+         x coordinate of reference position of slitless aperture
+    y_ref_slitless : float
+        y coordinate of reference position of slitless aperture
+    v2vert1 : float
+        slit vertex 1 in V2 frame
+    v2vert2 : float
+        slit vertex 2 in V2 frame
+    v2vert3 : float
+        slit vertex 3 in V2 frame
+    v2vert4 : float
+        slit vertex 4 in V2 frames
+    v3vert1 : float
+        slit vertex 1 in V3 frames
+    v3vert2 : float
+        slit vertex 2 in V3 frames
+    v3vert3 : float
+        slit vertex 3 in V3 frames
+    v3vert4 : float
+        slit vertex 4 in V3 frames
+    wavetable : numpy  2-D array
+        For each row in the slit hold  wavelength, and 
+        x center, ycenter, x and y box region cooresponding to the wavelength
+    """
+    schema_url = "http://stsci.edu/schemas/jwst_datamodel/specwcs_miri_lrs.schema"
+    reftype = "specwcs"
+
+    def __init__(self, init=None,
+                 wavetable=None,
+                 x_ref=None,
+                 y_ref=None,
+                 x_ref_slitless=None,
+                 y_ref_slitless=None,
+                 v2_vert1=None,
+                 v2_vert2=None,
+                 v2_vert3=None,
+                 v2_vert4=None,
+                 v3_vert1=None,
+                 v3_vert2=None,
+                 v3_vert3=None,
+                 v3_vert4=None,
+                 **kwargs):
+        super().__init__(init=init, **kwargs)
+
+        if init is None:
+            self.populate_meta()
+        if wavetable is not None:
+            self.wavetable = wavetable
+        if x_ref is not None:
+            self.meta.x_ref = x_ref
+        if y_ref is not None:
+            self.meta.y_ref = y_ref
+        if x_ref_slitless is not None:
+            self.meta.x_ref_slitless = x_ref_slitless
+        if y_ref_slitless is not None:
+            self.meta.y_ref_slitless = y_ref_slitless
+        if v2_vert1 is not None:
+            self.meta.v2_vert1 = v2_vert1
+        if v2_vert2 is not None:
+            self.meta.v2_vert2 = v2_vert2
+        if v2_vert3 is not None:
+            self.meta.v2_vert3 = v2_vert3
+        if v2_vert4 is not None:
+            self.meta.v2_vert4 = v2_vert4
+        if v3_vert1 is not None:
+            self.meta.v3_vert1 = v3_vert1
+        if v3_vert2 is not None:
+            self.meta.v3_vert2 = v3_vert2
+        if v3_vert3 is not None:
+            self.meta.v3_vert3 = v3_vert3
+        if v3_vert4 is not None:
+            self.meta.v3_vert4 = v3_vert4
+
+    def populate_meta(self):
+        self.meta.instrument.name = "MIRI"
+        self.meta.instrument.detector = "MIRIMAGE"
+        self.meta.reftype = self.reftype
```

Comment:
This needs more keywords to define a LRS fixed slit mode - filter?

Reply:
added  self.meta.instrument.filter = 'P750L'